packages/bpk-react-utils/readme.md

```diff
@@ -164,3 +164,48 @@ const MyComponent = (props) => (
 | appearClassName       | string   | true     | -             |
 | appearActiveClassName | string   | true     | -             |
 | transitionTimeout     | number   | true     | -             |
+
+## hasChildrenOfType.js
+
+A prop type checker that allows you to restrict a component's children to a specific type.
```

Comment:
Can we explain how this is different to `PropTypes.instanceOf`? That was my first question upon seeing this.

Reply:
Would adding 'Component' help? Totally open to suggestions if you have any ideas. 
eg.
> A prop type checker that allows you to restrict a component's children to a specific (Component) type.

Reply:
@janeklb I second @k0nserv's question, what's the difference between using `hasChildrenOfType(MyComponent)` & `PropTypes.instanceOf(MyComponent)`? I.e. i think we are questioning whether this new prop type is redundant or not.

Reply:
`PropTypes.instanceOf(...)` uses js's `instanceof` operator to see if the prop value is an instance of the supplied class/type. It doesn't work with components (ie. jsx or `React.createElement` etc.) since those can't be operated on with `instanceof`.

This type checker validates that the _children_ of a component are all "instances" of a certain component type.

Reply:
See the tests - they should give some more clarity :)

Reply:
tbh im not sure why this isn't a part of `prop-types` itself -- seems like a common enough requirement

Reply:
And airbnb's prop types seem to already support something like this https://www.npmjs.com/package/airbnb-prop-types